jina/math/dimensionality_reduction.py

```diff
@@ -0,0 +1,63 @@
+import numpy as np
+
+
+class PCA:
+    """A class for PCA dimensionality reduction in Jina."""
+
+    def __init__(self, n_components: int, whiten: bool = False):
+        """
+
+        :param n_components: Number of components to keep when projecting with PCA
+        :param whiten: Flag variable stating if there projecting with whitening
```

Comment:
you can remove docstring here and move to class docstrings, checkout how other classes write docstrings

Reply:
I will do it now